components/datetime/src/fields.rs

```diff
@@ -0,0 +1,353 @@
+use std::fmt;
+
+#[derive(Debug)]
+pub enum Error {
+    Unknown,
+    TooLong,
+}
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum FieldLength {
+    One = 1,
+    TwoDigit = 2,
+    Abbreviated = 3,
+    Wide = 4,
+    Narrow = 5,
+    Six = 6,
+}
+
+impl FieldLength {
+    pub fn try_from(idx: usize) -> Result<Self, Error> {
+        Ok(match idx {
```

Comment:
Nit: if this gets public then we probably need better names.
And the mapping between the numeric value and Abbreviate / Wide / Narrow is inconsistent in CLDR / ICU.

For example "x" and "X" (https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/text/SimpleDateFormat.html):
4: -0800 | -075258 | Z	
5: -08:00 | -07:52:58 | Z

Reply:
Personally, I would name these One, Two, Three, ... Six
They are not public, and the meaning of these is inconsistent in CLDR between various fields.

Reply:
Let's discuss the naming of the Fields when we work on skeletons! https://github.com/unicode-org/icu4x/issues/272